R/rdeephaven/lib/build-cpp.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+export SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+export CLIENT=$SCRIPT_DIR/cpp-client
+
+# First, we build all C++ dependencies
+# cd cpp-dependencies
+# chmod +x ./build-dependencies.sh
+# ./build-dependencies.sh
+# cd $SCRIPT_DIR
+
+# Next, we download, build, and install the Deephaven C++ client
+svn checkout https://github.com/deephaven/deephaven-core/trunk/cpp-client
```

Comment:
This looks like it is just pulling `main`.  Should this be versioned or locked to the SHA of the current branch or something?

Reply:
Is this really in subversion?!

Reply:
Try this:  https://git-scm.com/docs/git-sparse-checkout

https://medium.com/@marcoscannabrava/git-download-a-repositorys-specific-subfolder-ceeabc6023e2
https://github.blog/2020-01-17-bring-your-monorepo-down-to-size-with-sparse-checkout/

Reply:
Both of these problems need to be addressed in v2.